src/OpenTelemetry/Metrics/MetricReaderExt.cs

```diff
@@ -133,20 +133,22 @@ internal List<Metric> AddMetricsListWithViews(Instrument instrument, List<Metric
                     }
 
                     if (this.instrumentIdentityToMetric.TryGetValue(instrumentIdentity, out var existingMetric))
-                    {
-                        metrics.Add(existingMetric);
-                        continue;
-                    }
-
-                    if (this.metricStreamNames.Contains(metricStreamName))
                     {
                         OpenTelemetrySdkEventSource.Log.DuplicateMetricInstrument(
                             metricName,
                             meterName,
                             "Metric instrument has the same name as an existing one but differs by description, unit, or instrument type. Measurements from this instrument will still be exported but may result in conflicts.",
                             "Either change the name of the instrument or use MeterProviderBuilder.AddView to resolve the conflict.");
+
+                        if (metrics.Count == 0)
+                        {
+                            metrics.Add(existingMetric);
+                        }
+
+                        continue;
                     }
 
+                    // TODO: I think this check needs to be moved up...
```

Comment:
yes.

Reply:
Moved and added a test for this scenario https://github.com/open-telemetry/opentelemetry-dotnet/pull/3129/commits/462d1afe028c897d34bcc35f86f9e7b211bb474b